packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -580,6 +579,20 @@ export const Table = <
     );
   }, [data, getSubRows]);
 
+  const [rowHasParent] = React.useState(() => new Set<T>());
+
+  const getSubRowsWithSubComponents = React.useCallback(
+    (originalRow: T) => {
+      if (!rowHasParent.has(originalRow)) {
+        rowHasParent.add(originalRow);
```

Comment:
Since we are adding rows to this set and never removing them, this could grow indefinitely in size.

Consider using [`WeakSet`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet) instead, so that unused values are automatically garbage collected.